__tests__/app.test.js

```diff
@@ -125,4 +134,78 @@ describe("app.js", () => {
       expect(message).toBe("Invalid datatype for parameter");
     });
   });
+  describe("/api/articles/:article_id/comments", () => {
+    test("POST 201: responds to the client with an object containing the username and body of the comment", async () => {
+      const commentToPost = { username: "lurker", body: "I agree!" };
+      const response = await request(app)
+        .post("/api/articles/13/comments")
+        .send(commentToPost)
+        .expect(201);
+      const { body } = response;
+      const { comment } = body;
+      expect(Object.keys(comment).length).toBe(6);
+      expect(comment).toMatchObject({
+        comment_id: 19,
+        body: "I agree!",
+        article_id: 13,
+        author: "lurker",
+        votes: 0,
+      });
+      expect(typeof comment.created_at).toBe("string");
+      const result = await db.query(
+        `SELECT * FROM comments WHERE article_id = $1`,
+        [13]
+      );
+      expect(result.rows.length).toBe(1);
+    });
+    test("POST 404: responds to the client with an error message when the article id is valid but not found", async () => {
+      const commentToPost = { username: "lurker", body: "I agree!" };
+      const response = await request(app)
+        .post("/api/articles/1000/comments")
+        .send(commentToPost)
+        .expect(404);
+      const { body } = response;
+      const { message } = body;
+      expect(message).toBe("No article matching the provided ID");
+    });
+    test("POST 400: responds to the client with an error message when the article id is invalid", async () => {
+      const commentToPost = { username: "lurker", body: "I agree!" };
+      const response = await request(app)
+        .post("/api/articles/abc/comments")
+        .send(commentToPost)
+        .expect(400);
+      const { body } = response;
+      const { message } = body;
+      expect(message).toBe("Invalid datatype for parameter");
+    });
+    test("POST 400: responds to the client with an error message when the article_id exists but the username provided does not exist in the users table", async () => {
```

Comment:
It's really good that you included this test, but it would be considered a `404` rather than a `400`. Because the username only has to be a string it is _valid_ but it crucially _doesn't exist_